### PR TITLE
DAH-1229 switch to using markdown-to-jsx for processing markup

### PR DESCRIPTION
**Pull request by jacksongonzales**
[DAH-1229](https://sfgovdt.jira.com/jira/software/c/projects/DAH/boards/122?modal=detail&selectedIssue=DAH-1229)

This is also a proposal to switch to using markdown-to-jsx in favor of `dangerouslySettingInnerHTML`! I think we all know the detriments of `dangerouslySetInnerHTML`, and markdown-to-jsx handles our use cases nicely and avoids all the dangers of our previous approach. 🎉

### CODE REVIEW

**Review by kramduckner — APPROVED** [review]
lgtm

File: app/javascript/__tests__/components/__snapshots__/BeforeApplyingForSale.test.tsx.snap [hunk 1]
```diff
@@ -24,37 +24,50 @@ exports[`BeforeApplyingForSale display Before Applying when type is Habitat list
...
-      <div
-        dangerouslySetInnerHTML={
-          Object {
-            "__html": "Go to a <a href='https://habitatgsf.org/amber-drive-info/' target='_blank'>Habitat for Humanity information session</a>",
-          }
-        }
-      />
+      <div>
+        <span>
+          Go to a
```
Comment (christine-sfg) [inline app/javascript/__tests__/components/__snapshots__/BeforeApplyingForSale.test.tsx.snap:29]: This markup seem bloated. Is there any way we can clean this up?

File: app/javascript/__tests__/modules/listingDetails/__snapshots__/ListingDetailsEligibility.test.tsx.snap [hunk 2]
```diff
@@ -59,24 +59,27 @@ exports[`ListingDetailsLotteryInfo displays listing details eligibility section
...
             >
-              <span
-                dangerouslySetInnerHTML={
-                  Object {
-                    "__html": "For income calculations, household size includes everyone (all ages) living in the unit.",
-                  }
-                }
-              />
+              <span>
+                <span>
```
Comment (christine-sfg) [inline app/javascript/__tests__/modules/listingDetails/__snapshots__/ListingDetailsEligibility.test.tsx.snap:63]: here too

File: app/javascript/__tests__/modules/listingDetailsLottery/__snapshots__/ListingDetailsLotteryInfo.test.tsx.snap [hunk 3]
```diff
@@ -31,13 +31,60 @@ Array [
...
-            dangerouslySetInnerHTML={
-              Object {
-                "__html": "<p><span style=\\"font-size: 10pt; font-family: Arial; color: rgb(0, 0, 0); background-color: transparent;\\">The lottery will not be in person, but instead will be a virtual lottery.</span></p><p><span style=\\"font-size: 12pt; color: rgb(0, 0, 0); background-color: transparent;\\"> </span></p><p><span style=\\"font-size: 10pt; font-family: Arial; color: rgb(0, 0, 0); background-color: transparent;\\">Come back to this page on the day of the lottery for a link you can click on to watch the lottery live.</span></p><p><br></p><p><br></p>",
-              }
-            }
-          />
+          <div>
+            <span>
+              <p>
+                <span
```
Comment (christine-sfg) [inline app/javascript/__tests__/modules/listingDetailsLottery/__snapshots__/ListingDetailsLotteryInfo.test.tsx.snap:37]: hm, `p` inside a `span`

**Comment by jacksongonzales** [comment]
@christine-sfg thx for catching those; the plugin inserts a `<span>` by default so I modified those utils to actually be a block or an inline element when we want it to. We were also using inline in some spots where we probably didn't need to 😅 those are fixed now too

**Pull request merged**